builtin/logical/aws/rotation_test.go

```diff
@@ -437,3 +448,135 @@ func TestRequeueOnError(t *testing.T) {
+			SecretAccessKey: aws.String(fmt.Sprintf("%s-%d", *input.UserName, count)),
+		},
+	}, nil
+}
+
+// Test_RotationQueueInitialized creates a 2 node cluster and sets up the AWS
+// credentials backend. The test creates 3 sets of static credentials. Two of
+// those have a low rotation period and should get rotated during the test. The
+// third has a high rotation period and should not be rotated. The test verifies
+// that the correct secrets are rotated, then transfers leadership to the other
+// node. The test verifies that credentials are once again rotated on the new
+// active node.
+func Test_RotationQueueInitialized(t *testing.T) {
+	mockClient := &mockIAM{
+		newKeys: make(map[string]int),
+	}
+	coreConfig := &vault.CoreConfig{
+		LogicalBackends: map[string]logical.Factory{
+			"aws": func(ctx context.Context, config *logical.BackendConfig) (logical.Backend, error) {
+				b := Backend(config)
+				b.iamClient = mockClient
+				b.minAllowableRotationPeriod = 1 * time.Second
+				err := b.Setup(ctx, config)
+				return b, err
+			},
+		},
+		RollbackPeriod: 1 * time.Second,
+	}
+
+	cluster := vault.NewTestCluster(t, coreConfig, &vault.TestClusterOptions{
+		HandlerFunc: vaulthttp.Handler,
+		NumCores:    2,
+	})
+	cluster.Start()
+	defer cluster.Cleanup()
+
+	cores := cluster.Cores
+	vault.TestWaitActive(t, cores[0].Core)
+	client := cores[0].Client
+	err := client.Sys().Mount("aws", &api.MountInput{
+		Type: "aws",
+	})
+	require.NoError(t, err)
+
+	// create 3 static roles with different rotation periods
+	_, err = client.Logical().Write("aws/static-roles/role1", map[string]interface{}{
+		"username":        "user1",
+		"rotation_period": "2s",
+	})
+	require.NoError(t, err)
+	_, err = client.Logical().Write("aws/static-roles/role2", map[string]interface{}{
+		"username":        "user2",
+		"rotation_period": "1s",
+	})
+	require.NoError(t, err)
+	_, err = client.Logical().Write("aws/static-roles/role3", map[string]interface{}{
+		"username":        "user3",
+		"rotation_period": "5m",
+	})
+	require.NoError(t, err)
+
+	getSecret := func(c *api.Client, role string) string {
+		r, err := c.Logical().Read("aws/static-creds/" + role)
+		require.NoError(t, err)
+		return r.Data["secret_key"].(string)
+	}
+
+	role1Secret := getSecret(client, "role1")
+	role2Secret := getSecret(client, "role2")
+	role3Secret := getSecret(client, "role3")
+
+	verifySecretsRotated := func(c *api.Client, originalRole1Secret, originalRole2Secret, originalRole3Secret string) (updatedRole1Secret, updatedRole2Secret string) {
+		testhelpers.RetryUntil(t, 5*time.Second, func() error {
+			// verify that both secrets with a low rotation period get rotated
+			updatedRole1Secret = getSecret(c, "role1")
+			updatedRole2Secret = getSecret(c, "role2")
+
+			if originalRole1Secret == updatedRole1Secret && originalRole2Secret == updatedRole2Secret {
+				return fmt.Errorf("secrets haven't been rotated")
```

Comment:
nit: Is there a way to make this error point to the test file? Instead of the testhelpers.go file:

```
    testhelpers.go:726: did not complete before deadline, err: secrets haven't been rotated
```

Reply:
That's a good idea! I added a t.Helper() to the testhelpers function, so now we see:

```
    rotation_test.go:554: did not complete before deadline, err: secrets haven't been rotated

```